ui/src/utils/supabase.test.ts

```
@@ -0,0 +1,103 @@
+import { signInAnon, fetchConnections, addConnection, insertAnonConnection } from './supabase';
+
+const supabaseTestUUID: string = process.env.SUPABASE_TEST_UUID || '';
+
+let connectionsQty: number = 0;
+
+describe('fetchConnections', () => {
+    test('should fetch connections successfully', async () => {
+        try {
+            console.log('Starting fetchConnections...');
+            const result = await fetchConnections();
+            console.log('Result of testRequest:', result);
+            if (result) {
+                connectionsQty = result.length;
+                console.log('Number of connections:', connectionsQty);
+            } else {
+                console.log('No connections found');
+            }
+            console.log('fetchConnections completed successfully');
+        } catch (error) {
+            console.error('Error during fetchConnections execution');
+            throw error;
+        }
+    });
+});
+
+describe('addConnection', () => {
+    test('should add a connection successfully', async () => {
+        try {
+            console.log('Starting addConnection...');
+            const result = await addConnection(supabaseTestUUID);
+            if (!result.success) {
+                throw new Error(`Failed to add connection: ${result.error}`);
+            } else {
+                console.log('Connection added successfully:', result.data);
+                // Verify the connection was added
+                const connections = await fetchConnections();
+                if (connections) {
+                    expect(connections.length).toBe(connectionsQty + 1);
```

the `fetchConnections` query should probably be a count instead of returning rows, I think there's a default limit of 1k so eventually this test would fail due to pagination if we are testing with more realistic connection data. e.g. `.select('*', { count: 'exact' });`

---

this has been addressed in another branch which will be pulled in
- https://github.com/getlantern/unbounded/pull/273

`fetchConnections()` is now `connectionCount()`